### ISSUE TITLE
Horizontal window scrolling

### ISSUE THREAD

**Issue by GoogleCodeExporter** [mintty/mintty#138]
```
As far I know there is no way to disable wraping very long lines in mintty,
which I my opinion makes harder to fast analyze logging output (eg.
postgresql detailed log). Normally cygwin i.e. bash is running as cmd.exe
subprocess and display it's output there, where long lines could be
displayed without wraping. That is possible in cmd, when one set 'Screen
Buffer Size - width' to larger value than 'Window Size - width'.
Such a solution - add controlled buffer width size in addition to mintty
window size, will be very usefull, and will make possibility to prevent
long lines wrapping.

In addition to that, when buffer width > window width, 
and there is need to scroll window horizontally
one should be able to:
-disable/enable horizontal scroller
-scroll window horizontally by key combination
-scroll window horizontally by mouse whell + some enabler key eg. left alt
 (like in Adobe's apps - photoshop/illustrator etc.) 

```

Original issue reported on code.google.com by `dominik....@gmail.com` on 17 Oct 2009 at 6:18

**Comment by hamishmb** [comment]
Is there any more interest on this? I recognise that this isn't a priority, but it certainly would be a nice-to-have and using Mintty after other terminals is a bit jarring due to the lack of this feature.

**Comment by mintty** [comment]
First comment after 9 years, suggesting not much interest. And few other terminals have such a feature.
Like with tabs, I don't see it justified to spend significant effort and troublesome testing into a feature which is kind of exotic for terminals. However, again like with tabs, if you find out how a Windows window could be set up to apply a global view offset, so that could be a virtual implementation, we can talk about it.

**Comment by hamishmb** [comment]
Agreed. Mintty is a good terminal emulator, but things like eg a menubar so options are discoverable, and as above, a better buffer column width are things I consider essential in a modern terminal emulator.

In terms of other terminals not having this feature, every terminal I've used since around Ubuntu 10.04 has had it - it seems common to me.

**Comment by mintty** [comment]
> options are discoverable

The Options menu is available both from the title bar and the context menu.

**Comment by hamishmb** [comment]
But neither of those places are where one expects to find an options menu in Windows.

**Comment by BrianInglis** [comment]
What menus and options do you typically find in Unix terminals like 
`gnome-terminal`, `konsole`, `lxterminal`, `mate-terminal`, `qterminal`, `vte`, `xfce4-terminal`, `xterm` available on *Cygwin* and other distros? 
I expect only one or more none/`alt`/`ctrl`/`shift`-right/left button popups with *Options* or *Preferences* among the entries, maybe with an optional menu bar offering access to the same menus. 
Sometimes X resources are the only way to customize some terminal options. 
Options in `mintty` are so extensive that there really should be an HTML doc set for them. I am not sure I would prefer if they all were made available under either an *Office*-like set of screens or a browser-like list of config settings, as that would increase the code size and memory required, without really improving accessability or discoverability over the docs.

**Comment by BrianInglis** [comment]
Lengthy or wide output is best tee-d or redirected to a file and used in a viewer or editor where I can choose to scroll or wrap lines depending on whether I am dealing with linear or tabular content. 
This is Unix: "Use the tools, Luke." ;^>

**Comment by hamishmb** [comment]
Fair point about lengthy output, though having automatic line-rewrapping is useful.

Normally, I would expect to use Edit -> Preferences, or on Windowsey UIs, Tools -> Options. This is how mate-terminal behaves, and I imagine gnome-terminal uses the GNOME hamburger menu now, but honestly I'm not a fan of that UI.

I think the way the options are presented is okay, and not that different from anything else I've used, I'd just like it to be easier to find them.

**Comment by rgutmann** [comment]
> Lengthy or wide output is best tee-d or redirected to a file and used in a viewer or editor where I can choose to scroll or wrap lines depending on whether I am dealing with linear or tabular content. This is Unix: "Use the tools, Luke." ;^>

I'm not saying that I haven't been able to deal with this before I found the solution which I'm currently practicing, but to be honest with you: this sounds a lot like "blame the victim" type of argument.

All I meant was that it is much nicer, easier and faster to be able to configure the terminal to have a wider window buffer with a horizontal scrollbar. So we could try to improve the user experience instead of asking him to jump through loops, couldn't we?

**Comment by hamishmb** [comment]
Agreed. Perhaps especially for Windows users who might not be familiar with the UNIXey tools and way of doing things ie Cygwin and mintty might be their first Unix/Linux-like environment. They're more likely to get confused and frustrated that Linux/Unix users

That said, mintty is still a great terminal, and I don't know what I'd do without it - I certainly don't want to imply that I'm not grateful for all the work that's been put into it.

**Comment by mintty** [comment]
Released 3.5.3.

**Comment by hamishmb** [comment]
Thanks for implementing this!
